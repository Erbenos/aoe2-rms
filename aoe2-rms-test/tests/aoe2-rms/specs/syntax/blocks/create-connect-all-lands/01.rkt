#lang aoe2-rms
; Interconnect all player islands and neutral islands.

<LAND-GENERATION>
(base-terrain 'WATER)
(create-player-lands
 (terrain-type 'DESERT)
 (number-of-tiles 100))
(create-land
 (terrain-type 'FOREST)
 (number-of-tiles 100)
 (land-position 99 1))
(create-land
 (terrain-type 'PINE_FOREST)
 (number-of-tiles 100)
 (land-position 50 50))

<CONNECTION-GENERATION>
(create-connect-all-lands (replace-terrain 'WATER 'SHALLOW))
