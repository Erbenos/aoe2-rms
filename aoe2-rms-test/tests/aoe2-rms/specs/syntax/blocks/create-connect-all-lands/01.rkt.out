<LAND_GENERATION>
base_terrain WATER
create_player_lands {
  terrain_type DESERT
  number_of_tiles 100
}
create_land {
  terrain_type FOREST
  number_of_tiles 100
  land_position 99 1
}
create_land {
  terrain_type PINE_FOREST
  number_of_tiles 100
  land_position 50 50
}
<CONNECTION_GENERATION>
create_connect_all_lands {
  replace_terrain WATER SHALLOW
}
